components/tidb_query_datatype/src/codec/row/v2/row_slice.rs

```diff
@@ -69,11 +71,11 @@ impl RowSlice<'_> {
                 if let Ok(idx) = non_null_ids.binary_search(&(id as u32)) {
                     let offset = offsets.get(idx).ok_or(Error::ColumnOffset(idx))?;
                     let start = if idx > 0 {
-                        offsets[idx - 1] as usize
+                        offsets.get_unchecked(idx - 1) as usize
```

Comment:
There might not be sufficient elements in the offsets (if the data is corrupted). We'd better handle it.

Previously this case result in panic. Now it result in memory issue, which is worse. Better to change it to return error.

Reply:
I see this conversation marked as resolved, but what was the resolution? Why is this unchecked indexing ok?

Reply:
Doing an unsafe array access here isn't obviously important to performance. @zhongzc are you sure this needs to be unchecked?

Reply:
@brson  Previous `offsets.get(idx)` indicates `offsets.get_unchecked(idx - 1)` is okay.